### DISCUSSION TITLE
Retrieve resource from theme

### DISCUSSION THREAD

**Discussion by moving-bits** [mapsforge/vtm#967] [Q&A] [answered]
I'm working on implementing a theme legend view for VTM themes (as I have done already based on Mapsforge, using a grid layout within a bottom sheet dialog).

How can I retrieve a graphics resource from theme as bitmap?
For Mapsforge this could be done using `XMLUtils.createBitmap()`, I'm looking for something similar.

Thanks.

**Comment by devemux86** [comment]
Something similar is `CanvasAdapter.getBitmapAsset`.

However these classes and methods are for internal use.
For external use better host any images somewhere else.

**Reply by moving-bits** [reply]
Thanks for the quick reply. 

Self-hosting all theme images isn't really an option, IMHO, as we would have to include hundreds of additional pics to our APK, although they are already present in the theme, not only increasing our APK size significantly, but also a nightmare to maintain on theme updates.

I had found `CanvasAdapter.createBitmap()`, which looked somewhat similar to me, but that method is marked `protected`. Would it be ok to expose this method as `public` (in case that it actually is what I'm looking for, haven't tested yet)? (I would test this first, and prepare a PR on successful test then.)

I'm also looking into `CanvasAdapter.getBitmapAsset()` (which is `public` already), but I'm stumbling about how to convert a `org.oscim.backend.canvas.Bitmap` into a `android.graphics.Bitmap` which is what `ImageView.setImageBitmap()` expects. (Edit: Mapsforge's AndroidBitmap has a public member `bitmap` for this, whereas VTM's AndroidBitmap member `mBitmap` is not accessible from the outside.)

**Comment by devemux86** [comment, marked as answer]
`CanvasAdapter` has implementations per platform.

> Something similar is `CanvasAdapter.getBitmapAsset`.

> `org.oscim.backend.canvas.Bitmap` into a `android.graphics.Bitmap`

`AndroidGraphics.getBitmap`

> Mapsforge's AndroidBitmap has a public member bitmap for this

`AndroidGraphicFactory.getBitmap`

**Reply by moving-bits** [reply]
I was able to solve my issue by using `setImageBitmap(AndroidGraphics.getBitmap(CanvasAdapter.getBitmapAsset(relativePath, filename, resourceProvider, 0, height, 100)));` - Thanks for the pointers!
